### ISSUE TITLE
jetty.sh does not stop jetty anymore

### ISSUE THREAD

**Issue by l-cdc** [eclipse/jetty.project#10271]
**Jetty version(s)**
10.0.15

**Java version/vendor** `(use: java -version)`
openjdk version "11.0.19" 2023-04-18 LTS
OpenJDK Runtime Environment (Red_Hat-11.0.19.0.7-1.el7_9) (build 11.0.19+7-LTS)
OpenJDK 64-Bit Server VM (Red_Hat-11.0.19.0.7-1.el7_9) (build 11.0.19+7-LTS, mixed mode, sharing)

**OS type/version**
Red Hat Enterprise Linux Server release 7.9 (Maipo)

**Description**
This bug was likely introduced by #9309 which introduces `xargs` to launch `java`. The [relevant lines](https://github.com/eclipse/jetty.project/blob/91455908205a85c68582b1aa01baaae527c5eef3/jetty-home/src/main/resources/bin/jetty.sh#L502) are

```sh
        echo ${RUN_ARGS[*]} | xargs ${JAVA} > /dev/null &
        disown $!
        echo $! > "$JETTY_PID"
```
`$!` now returns the PID of the `xargs` process, not the java one. Consequently, `jetty.sh stop` now stops `xargs`, but at least on my system killing `xargs` (version "xargs (GNU findutils) 4.5.11") does not terminate its child processes.

Note also that on systems with `start-stop-daemon`, this probably works as `start-stop-daemon` writes the child PID in the PID file if I understand the man page correctly.

The probably simplest (not best) way to fix this is to retrieve the correct PID with something like
```sh
 echo `pgrep -P $!` > "$JETTY_PID"
```

**How to reproduce?**
Issue should appear with unmodified jetty distribution on any system without `start-stop-daemon`.

**Comment by joakime** [comment]
Thank you for the detailed issue, and initial triage.

I think having Jetty itself produce the pid file is going to be the safest solution.

**Comment by joakime** [comment]
Opened PR #10272  - @l-cdc can you give try out that branch?  see if it works for you?
Meanwhile, I'll continue testing ...

**Comment by joakime** [comment]
@l-cdc what kind of environment are you running in which doesn't have `start-stop-daemon` ?
I'm having a hard time finding one to test in.

**Comment by joakime** [comment]
Ah! there we go ...

```
$ docker run --rm -it redhat/ubi8:latest bash
Unable to find image 'redhat/ubi8:latest' locally
latest: Pulling from redhat/ubi8
0fa65fe5c23e: Pull complete 
Digest: sha256:a7143118671dfc61aca46e8ab9e488500495a3c4c73a69577ca9386564614c13
Status: Downloaded newer image for redhat/ubi8:latest
[root@79aca762789a /]# which start-stop-daemon
/usr/bin/which: no start-stop-daemon in (/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin)
[root@79aca762789a /]# 
```

**Comment by l-cdc** [comment]
@joakime I tested the branch `fix/10.0.x/start-pidfile` and it works for me. Many thanks for the very rapid response and PR!
